app/backend/core/authentication.py

```diff
@@ -0,0 +1,164 @@
+# Refactored from https://github.com/Azure-Samples/ms-identity-python-on-behalf-of
+
+from quart import request
+from abc import ABC
+import msal, os
+from msgraph import GraphServiceClient
+from azure.core.credentials import TokenCredential, AccessToken
+import logging
+from typing import Any
+
+class AuthError(Exception):
+    error: str = None
+    status_code: int = None
+
+    def __init__(self, error, status_code):
+        self.error = error
+        self.status_code = status_code
+
+# TokenCredential wrapper class to use an auth token to call the Microsoft Graph API
+class AuthToken(TokenCredential):
+    _token: dict = None
+
+    def __init__(self, token: dict):
+        self._token = token
+
+    def get_token(self, *scopes, **kwargs) -> AccessToken:
+        return AccessToken(self._token["access_token"], self._token["expires_in"])
+
+
+class AuthenticationHelper(ABC):
+
+    _confidential_client: msal.ConfidentialClientApplication = None
+    _authority: str = "https://login.microsoftonline.com/{}".format(os.getenv("AZURE_TENANT_ID"))
+
+    @staticmethod
+    def use_authentication():
+        # Checks for environment variables related to authentication.
+        # If they are not defined, authentication is not used
+        var_names = ["AZURE_USE_AUTHENTICATION", "AZURE_SERVER_APP_ID", "AZURE_SERVER_APP_SECRET", "AZURE_CLIENT_APP_ID", "AZURE_TENANT_ID"]
+        return all(var_name in os.environ for var_name in var_names)
+
+    @staticmethod
+    def get_auth_setup_for_client() -> dict[str, Any]:
+        # returns MSAL.js settings used by the client app
+        return {
+            "useLogin": AuthenticationHelper.use_authentication(), # Whether or not login elements are enabled on the UI
+            "msalConfig": {
+                "auth": {
+                    "clientId": os.getenv("AZURE_CLIENT_APP_ID"), # Client app id used for login
```

Comment:
This code references a lot of env variables, which makes it harder to test. Could all configuration be passed in via class constructors or function args instead?